tests/LogOtter.CosmosDb.ContainerMock.IntegrationTests/CosmosQueryEquivalencyTests.cs

```diff
@@ -496,6 +496,56 @@ await _testCosmos.GivenAnExistingItem(
         testResults!.SingleOrDefault().Should().BeEquivalentTo(realResults!.SingleOrDefault());
     }
 
+    [Fact]
+    public async Task GivenAQueryUsingStringEqualsMethodThenBothShouldWork()
+    {
+        await _testCosmos.GivenAnExistingItem(
+            new TestModel
+            {
+                Id = "RECORD1",
+                Name = "Bob Bobertson",
+                Value = false,
+                PartitionKey = "partition"
+            }
+        );
+
+        var (realResults, testResults) = await _testCosmos.WhenExecutingAQuery<TestModel>(
+            "partition",
+            q => q.Where(tm => tm.Name != null && tm.Name.Equals("Bob Bobertson"))
+        );
+
+        realResults.Should().NotBeNull();
+        testResults.Should().NotBeNull();
+
+        realResults!.SingleOrDefault().Should().NotBeNull();
+        testResults!.SingleOrDefault().Should().BeEquivalentTo(realResults!.SingleOrDefault());
+    }
+
+    [Fact]
+    public async Task GivenAQueryUsingStringEqualsMethodWithInvariantCultureIgnoreCaseThenBothShouldWork()
+    {
+        await _testCosmos.GivenAnExistingItem(
+            new TestModel
+            {
+                Id = "RECORD1",
+                Name = "Bob Bobertson",
+                Value = false,
+                PartitionKey = "partition"
+            }
+        );
+
+        var (realResults, testResults) = await _testCosmos.WhenExecutingAQuery<TestModel>(
+            "partition",
+            q => q.Where(tm => tm.Name != null && tm.Name.Equals("bob bobertson", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Does Cosmos now support any comparison passed to it? If it doesn't support some of the others in `StringComparison` it is probably worth adding a few negative tests and ensuring `ContainerMock` throws the same exception

Reply:
I've added a few more tests